compliance/aws/ecs_unused/AWS_Unused_ECS_Clusters.pt

```diff
@@ -0,0 +1,399 @@
+
+#Generates list of Regions.
+datasource "ds_regions_list" do
+  run_script $js_regions_map
+end
+
+#To get list of All ESC Clusters.
+datasource "ds_aws_ecs_cluster_list" do
+  iterate $ds_regions_list
+  request do
+    run_script $js_aws_ecs_cluster_list, val(iter_item,"region")
+  end
+  result do
+    encoding "json"
+    field "region", val(iter_item,"region")
+	field "clusterArns", jmes_path(response, "clusterArns")
+  end
+end
+
+#To get description of all clusters by passing array of cluster_name as input. 
+datasource "ds_aws_ecs_cluster_description" do
+  iterate $ds_regions_list
+  request do
+    run_script $js_aws_ecs_cluster_description, $ds_aws_ecs_cluster_list, val(iter_item,"region")
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "clusters[*]") do
+      field "cluster_Name", jmes_path(col_item, "clusterName")
+      field "active_Services_Count", jmes_path(col_item, "activeServicesCount")
+      field "running_Tasks_Count",  jmes_path(col_item, "runningTasksCount")
+      field "pending_Tasks_Count",  jmes_path(col_item, "pendingTasksCount")
+      field "registered_Container_Instances_Count", jmes_path(col_item, "registeredContainerInstancesCount")
+      field "status", jmes_path(col_item,"status")
+      field "region", val(iter_item,"region")
+      field "tags" do
+        collect jmes_path(col_item,"tags") do
+          field "tagKey", jmes_path(col_item,"key")
+          field "tagValue", jmes_path(col_item,"value")
+        end
+      end
+    end
+  end
+end
+
+datasource "ds_unused_ecs_cluster_map" do
+  run_script $js_aws_ecs_cluster_filter_map, $ds_aws_ecs_cluster_description, $param_exclude_tags
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+script "js_regions_map", type: "javascript" do
+  result "regions_map"
+  code <<-EOS
+    var regions_map=[]
+    regions_map.push({"region": "us-east-1"}, {"region": "us-east-2"}, {"region": "us-west-1"}, {"region": "us-west-2"}, {"region": "ap-south-1"}, {"region": "ap-northeast-2"}, {"region": "ap-southeast-1"}, {"region": "ap-southeast-2"}, {"region": "ap-northeast-1"}, {"region": "ca-central-1"}, {"region": "eu-central-1"}, {"region": "eu-west-1"}, {"region": "eu-west-2"}, {"region": "eu-west-3"}, {"region": "sa-east-1"}, {"region": "eu-north-1"})
+  EOS
+end
+
+#https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ListClusters.html
+script "js_aws_ecs_cluster_list", type: "javascript" do
+  parameters "region"
+  result "results"
+  code <<-EOS
+    results = {
+      auth: "auth_"+region.replace(/-/g,"_"),
+      host: 'ecs.'+region+'.amazonaws.com',
+      path: '/',
+      verb: 'POST',
+      headers: {
+        "X-Amz-Target": "AmazonEC2ContainerServiceV20141113.ListClusters",
+        "Content-Type": "application/x-amz-json-1.1",
+      },
+      body_fields: {"empty": "empty"}
```

Comment:
Can you explain the body_fields value.  I don't see empty: empty in the api docs.

Reply:
there should be an empty body('{ }') as part of API request or else we will be getting 400 - bad request.
Since we can not have  empty body_field('{ }') in the API request syntax, So I included  {"empty": "empty"}